### PR TITLE
Create initial doxygen base for the CSDK Hub

### PR DESCRIPTION
**Pull request by sarenameas**
- Doxygen will need to be run in the coreMQTT repo first to generate the TAG file:
  ```
  cd libraries/standard/coreMQTT 
  doxygen docs/doxygen/config.doxyfile  
  ```
- Run Doxygen in the CSDK repo with:
  ```
  doxygen docs/doxygen/config.doxyfile  
  ```
### Changes:
- Renamed doc to docs to match official for the child repos. 
- Took templates from doxygen_templates and updated FIXMEs.
- Add link to coreMQTT tag file. See https://github.com/FreeRTOS/coreMQTT/pull/32
- Added template page.txt for others to get started adding thing.

### In future PRs:
- Any scripts for CI, to run the doxygen of all children repos then this one, is not done in this PR and should be in a separate PR.
- The content of the main CSDK page (demos, migration, porting guides, and style guide) is upcoming in separate PRs from other team members.
- Fixing of warnings in the platform/include folder will be in a separate PR.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/doxygen/config.doxyfile [hunk 1]
```diff
@@ -0,0 +1,2565 @@
...
+# Adding location for the tag files is done as follows:
+# TAGFILES = file1=loc1 "file2 = loc2" ...
+# where loc1 and loc2 can be relative or absolute paths or URLs. See the
+# section "Linking to external documentation" for more information about the use
+# of tag files.
+# Note: Each tag file must have a unique name (where the name does NOT include
+# the path). If a tag file is not located in the directory in which doxygen is
+# run, you must also specify the path to the tagfile here.
+
+TAGFILES               = libraries/standard/coreMQTT/docs/doxygen/output/mqtt.tag=../../../../libraries/standard/coreMQTT/docs/doxygen/output/html
```
Comment (sarenameas) [inline docs/doxygen/config.doxyfile:2228]: Tags to child repos here.

File: docs/doxygen_guide.md [hunk 2]
```diff
@@ -98,24 +98,24 @@ so that Doxygen can copy the function signature to custom function pages.
...
 
-1. Please copy the all of the template pages to your library's doc/doxygen folder.
-For our CI to detect files correctly, please keep the folder name as *doc/doxygen*
+1. Please copy the all of the template pages to your library's docs/doxygen folder.
+For our CI to detect files correctly, please keep the folder name as *docs/doxygen*
 and all of the files names the same.
 
     ```console
-    cp -R <csdk_root>/doc/doxygen_templates <library_root>/doc/doxygen
+    cp -R <csdk_root>/docs/doxygen_templates <library_root>/docs/doxygen
```
Comment (yourslab) [inline docs/doxygen_guide.md:106]: I think this should be `config.doxyfile`

File: docs/doxygen_guide.md [hunk 3]
```diff
@@ -98,24 +98,24 @@ so that Doxygen can copy the function signature to custom function pages.
...
     ```
 
-1. Search for "FIXME"s in <library_root>/doc/doxygen/config.doxygen and update with
+1. Search for "FIXME"s in <library_root>/docs/doxygen/config.doxygen and update with
```
Comment (yourslab) [inline docs/doxygen_guide.md:109]: I guess we are using docs instead of doc for the libraries too?

Comment (sarenameas) [reply docs/doxygen_guide.md:109]: Yes

File: docs/doxygen_guide.md (hunk 2 above)
Comment (sarenameas) [reply docs/doxygen_guide.md:106]: No this is copying all the files in the directory. Let me add the missing *.

**Review by yourslab — DISMISSED** [review]

**Review by muneebahmed10 — DISMISSED** [review]

**Review by n9wxu — APPROVED** [review]

**Review by yourslab — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]

**Pull request merged**